From: Stephen Rothwell <sfr@canb.auug.org.au>
Date: Wed, 15 Feb 2017 11:30:03 +1100
Subject: RDMA/bnxt_re: fix for "bnxt_en: Update to firmware interface spec
 1.7.0."
Patch-mainline: v4.11-rc1
Git-commit: db690328a7df0b507f7d59de0c7e1bbe8f4b9e6a
References: bsc#1020413, FATE#321905

When the firmware interface spec was updated, a constant element was
renamed.  The rename missed the instances in the bnxt_re driver
because it wasn't upstream yet.  This updates the bnxt_re driver
with the rename.

Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/infiniband/hw/bnxt_re/main.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/infiniband/hw/bnxt_re/main.c
+++ b/drivers/infiniband/hw/bnxt_re/main.c
@@ -228,7 +228,7 @@ static int bnxt_re_net_ring_free(struct
 	}
 
 	bnxt_re_init_hwrm_hdr(rdev, (void *)&req, HWRM_RING_FREE, -1, -1);
-	req.ring_type = RING_ALLOC_REQ_RING_TYPE_CMPL;
+	req.ring_type = RING_ALLOC_REQ_RING_TYPE_L2_CMPL;
 	req.ring_id = cpu_to_le16(fw_ring_id);
 	bnxt_re_fill_fw_msg(&fw_msg, (void *)&req, sizeof(req), (void *)&resp,
 			    sizeof(resp), DFLT_HWRM_CMD_TIMEOUT);
@@ -268,7 +268,7 @@ static int bnxt_re_net_ring_alloc(struct
 	/* Association of ring index with doorbell index and MSIX number */
 	req.logical_id = cpu_to_le16(map_index);
 	req.length = cpu_to_le32(ring_mask + 1);
-	req.ring_type = RING_ALLOC_REQ_RING_TYPE_CMPL;
+	req.ring_type = RING_ALLOC_REQ_RING_TYPE_L2_CMPL;
 	req.int_mode = RING_ALLOC_REQ_INT_MODE_MSIX;
 	bnxt_re_fill_fw_msg(&fw_msg, (void *)&req, sizeof(req), (void *)&resp,
 			    sizeof(resp), DFLT_HWRM_CMD_TIMEOUT);
